kura/examples/test/org.eclipse.kura.example.wire.bool.multiport.provider.test/src/main/java/org/eclipse/kura/example/wire/bool/multiport/provider/test/BoolLogicTest.java

```diff
@@ -0,0 +1,473 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.example.wire.bool.multiport.provider.test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.configuration.ComponentConfiguration;
+import org.eclipse.kura.configuration.ConfigurationService;
+import org.eclipse.kura.type.TypedValue;
+import org.eclipse.kura.type.TypedValues;
+import org.eclipse.kura.util.wire.test.GraphBuilder;
+import org.eclipse.kura.util.wire.test.TestEmitterReceiver;
+import org.eclipse.kura.util.wire.test.WireTestUtil;
+import org.eclipse.kura.wire.WireEnvelope;
+import org.eclipse.kura.wire.WireRecord;
+import org.eclipse.kura.wire.graph.WireGraphConfiguration;
+import org.eclipse.kura.wire.graph.WireGraphService;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.LoggerFactory;
+
+public class BoolLogicTest {
+
+    // See:
+    // http://stackoverflow.com/questions/7161338/using-osgi-declarative-services-in-the-context-of-a-junit-test
+
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(BoolLogicTest.class);
+
+    private static final String UNDER_TEST_PID = "under.test";
+    private static final String TEST_FIRST_EMITTER_PID = "test.first.emitter.pid";
+    private static final String TEST_SECOND_EMITTER_PID = "test.second.emitter.pid";
+    private static final String TEST_RECEIVER_PID = "test.receiver.pid";
+
+    // in ports
+    private static final int IN0_PORT = 0;
+    private static final int IN1_PORT = 1;
+
+    // out ports
+    private static final int OUT_PORT = 0;
+
+    // configuration properties of component under test
+    private static final String FIRST_OPERAND_NAME_PROP_NAME = "operand.name.1";
+    private static final String SECOND_OPERAND_NAME_PROP_NAME = "operand.name.2";
+    // private static final String RESULT_NAME_PROP_NAME = "result.name";
+    private static final String BARRIER_MODALITY_PROPERTY_KEY = "barrier";
+    private static final String BOOLEAN_OPERATION = "boolean.operation";
```

Comment:
After the last refactor the tests should fail

Reply:
noted.